plugin/al/lib/AL_USDMaya/AL/usdmaya/SelectabilityDB.h

```diff
@@ -28,23 +29,60 @@ namespace usdmaya {
 ///---------------------------------------------------------------------------------------------------------------------
 /// \brief  Logic that stores a sorted list of paths which represent Selectable points in the USD hierarchy
 ///---------------------------------------------------------------------------------------------------------------------
-class SelectabilityDB {
+class SelectabilityDB
+{
+  // at little unpleasant design wise, however it means the proxy shape can directly modify the internal 
+  // list of unselectable paths. I have looked at moving some of that logic into this class, however that would
+  // incur a performance penalty (lock & excluded prim processing can be done at the same time).
+  friend class nodes::ProxyShape;
 public:
 
+  SelectabilityDB()
+    : m_unselectablePaths()
+  {
+    m_unselectablePaths.reserve(128);
+  }
+
   ///-------------------------------------------------------------------------------------------------------------------
   /// \brief  Determines this path is unselectable
   /// \param  path that you want to determine if it's unselectable
   ///-------------------------------------------------------------------------------------------------------------------
   AL_USDMAYA_PUBLIC
   bool isPathUnselectable(const SdfPath& path) const;
 
+  ///-------------------------------------------------------------------------------------------------------------------
+  /// \brief  Determines whether there is an internal entry for the path specified (and only this path!). If you wish 
+  ///         to determine selectability, call isPathUnselectable instead.
+  /// \param  path the path to check to see if exists
+  /// \return true if the path is contained, false if not.
+  ///-------------------------------------------------------------------------------------------------------------------
+  bool containsPath(const SdfPath& path) const
+  {
+    auto foundPathEntry = std::lower_bound(m_unselectablePaths.begin(), m_unselectablePaths.end(), path);
+    if(foundPathEntry != m_unselectablePaths.end() && path == *foundPathEntry)
```

Comment:
Nit-pick: could just return the result of the boolean expression and avoid the conditional altogether.

return (foundPathEntry != m_unselectablePaths.end() && path == *foundPathEntry);

Fine as it is, of course.